ERCS/erc-7798.md

```diff
@@ -0,0 +1,200 @@
+- An optional mechanism for relaying large JSON payloads using a backend relayer
+
+
+```mermaid
+sequenceDiagram
+    Sender->>PoS: Generate checkout intent
+    PoS->>NFC Relayer: Upload checkout intent JSON
+    PoS->>JSON RPC: requestContactlessPayment(uuid)
+    JSON RPC->>Wallet: Trigger NFC HCE
+    Recipient->>Wallet: Scan NFC Tag
+    Wallet->>Recipient: Checkout Intent URI
+    Recipient->>NFC Relayer: requestPaymentParams
+    NFC Relayer->>Recipient: Transaction data
+    Recipient->>Recipient: signTransaction
+    Recipient->>RPC Proxy: submitTransaction
+```
+
+
+### Use Cases
+
+#### Use Case 1: Customer purchases from a merchant
+1. A customer approaches a merchant and orders an item at their cash register (e.g. a cup of coffee).
+1. The merchant is running an app on their point-of-sale (PoS) system that allows initiating a checkout flow. 
+1. When the customer is done ordering, the merchant goes through the checkout flow up and triggers a payment request. 
+1. The merchant app calls the `requestContactlessPayment` RPC method with the expected payload. 
+1. The merchant's device emits an NFC signal via host card emulation (HCE) for the customer’s device to read.
+1. The customer, running their wallet software of choice, reads the NFC signal. The customer’s wallet constructs the necessary transaction(s) to execute the onchain checkout. 
+1. The wallet then signs and submits all transactions.
+1. Upon successful onchain execution of these transactions, the checkout is complete.
+
+#### Use Case 2: Peer-to-peer Sends
+1. Alice requests a payment on her wallet app by emitting an NFC signal. The contents of the signal is an [ERC-681](erc-681.md) URI
+> []will this only work with 681? 681 has limitations, like receiver intent which would be great to solve too.
+
+1. [ERC-681](erc-681.md) URI Bob reads the NFC signal from Alice on his wallet app and constructs the transaction
+2. Bob holds his phone close to Alice's which triggers the NFC flow.
+1. Bob then signs and submits the transaction onchain.
+1. Send flow is complete.
+
+### requestContactlessPayment
+The requestContactlessPayment method will be added as a standard method to the [EIP-1193 Ethereum Provider JavaScript API](eip-1193). 
+
+This method takes in two parameters: `type` and `uri`.
+
+- `type` is an enum that represents the payload type:
+    - 1 means the URI is an ERC-681-compliant payload.
+    - 2 means the URI points to an HTTP relayer, where the response of the relayer is either a JSON object with the transaction data already encoded as a 0x string, or a message for the customer to sign. The customer wallet is responsible for constructing, signing and submitting the transaction/message.
+
+- `uri` represents either the ERC-681-compliant payload or the relayer endpoint to query for the JSON payload. 
+
+Example calls:
+
+```js 
+window.ethereum.request({ 
+  method: 'requestContactlessPayment', 
+  params: { 
+    type: 1,
```

Comment:
Suggest using strings for type instead of numbers. Specifically, type here should be `erc681` and then for the relay perhaps a separate ERC could be defined for the relay design. This allows easier future expansion to more types of in-NFC payloads and relayers.